Source/WTF/wtf/CrossThreadCopier.h

```diff
@@ -92,12 +92,21 @@ template<typename T>
 struct CrossThreadCopierBase<false, false, T> {
     using Type = T;
     static constexpr bool IsNeeded = HasIsolatedCopy<T>::value;
-    template<typename U> static auto copy(U&& value) -> Type
+
+    template<typename U> static Type copy(const U& value)
```

Comment:
How would this change help catch the issue? It seems to me like it will still implicitly copy if no isolatedCopy() is available? Shouldn't we require an `isolatedCopy()` to be available if `crossThreadCopy()` gets called on a non-built-in type?

Reply:
> Do you think we should require isolatedCopy() on all non-builtin types? Might be a bigger change but I can investigate if it's preferred

Can you take a look to see how big of a change? I suspect it wouldn't be *that* big but not 100% sure.
I still feel this API is very error-prone. Given how hard these kind of bugs are to investigate, I think it is worth the effort.